dev/MRTCore/mrt/Microsoft.ApplicationModel.Resources/src/Microsoft.ApplicationModel.Resources.idl

```diff
@@ -54,6 +54,10 @@ namespace Microsoft.ApplicationModel.Resources
         IKeyValuePair<String, ResourceCandidate> GetValueByIndex(UInt32 index);
         [method_name("GetValueByIndexWithContext")]
         IKeyValuePair<String, ResourceCandidate> GetValueByIndex(UInt32 index, ResourceContext context);
+
+        ResourceCandidate TryGetValue(String resource);
```

Comment:
Should we instead make this `bool TryGetValue(String resource, out ResourceCandidate)` in order to fit the general pattern for TryGet* APIs (e.g. https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-5.0)?

Reply:
According to https://osgwiki.com/wiki/WinRT_API_Design_Patterns#Try_Pattern, we may return null or false. @ptorr-msft , any opinion?

Reply:
What @huichen123 has is the preferred form for methods that return reference types.  "bool+out" pairs are for value types.  This change avoids a behavioral-breaking change while adding the "don't throw if not found" behavior.

Reply:
The way I read the guidance, you should be using the Boolean pattern, as suggested.  But I'm not expert on this pattern so I defer to @jonwis.  (I would do the bool TryGetValue form, it seems to match most of the existing examples).

Reply:
Jon just approved the PR, so I guess he has spoken. :-)

Reply:
Cool, I learned something new today! :)